xray/services/report.go

```diff
@@ -83,22 +91,71 @@ type ReportCve struct {
+type LicensesReportRequestParams struct {
+	Name      string         `json:"name,omitempty"`
+	Filters   LicensesFilter `json:"filters,omitempty"`
+	Resources Resource       `json:"resources,omitempty"`
+}
+
+// ViolationsReportRequestParams defines a report request
+type ViolationsReportRequestParams struct {
+	Name      string           `json:"name,omitempty"`
+	Filters   ViolationsFilter `json:"filters,omitempty"`
+	Resources Resource         `json:"resources,omitempty"`
+}
+
+type VulnerabilitiesFilter struct {
+	VulnerableComponent string        `json:"vulnerable_component,omitempty"`
+	ImpactedArtifact    string        `json:"impacted_artifact,omitempty"`
+	SummaryContains     string        `json:"summary_contains,omitempty"`
+	HasRemediation      *bool         `json:"has_remediation,omitempty"`
+	Cve                 string        `json:"cve,omitempty"`
+	IssueId             string        `json:"issue_id,omitempty"`
+	Severity            []string      `json:"severities,omitempty"`
+	CvssScore           CvssScore     `json:"cvss_score,omitempty"`
+	Published           DateTimeRange `json:"published,omitempty"`
+	ScanDate            DateTimeRange `json:"scan_date,omitempty"`
 }
 
-type Filter struct {
-	HasRemediation *bool     `json:"has_remediation,omitempty"`
-	CvssScore      CvssScore `json:"cvss_score,omitempty"`
-	Severity       []string  `json:"severities,omitempty"`
+type DateTimeRange struct {
+	Start string `json:"start,omitempty"`
+	End   string `json:"end,omitempty"`
+}
+
+type LicensesFilter struct {
+	Component       string        `json:"component,omitempty"`
+	Artifact        string        `json:"artifact,omitempty"`
+	Unknown         *bool         `json:"unknown,omitempty"`
+	Unrecognized    *bool         `json:"unrecognized,omitempty"`
+	LicenseNames    []string      `json:"license_names,omitempty"`
+	LicensePatterns []string      `json:"license_patterns,omitempty"`
+	ScanDate        DateTimeRange `json:"scan_date,omitempty"`
+}
+
+type ViolationsFilter struct {
+	Type            string                `json:"type,omitempty"`
+	WatchNames      string                `json:"watch_names,omitempty"`
+	WatchPatterns   string                `json:"watch_patterns,omitempty"`
+	Component       string                `json:"component,omitempty"`
+	Artifact        string                `json:"artifact,omitempty"`
+	PolicyNames     []string              `json:"policy_names,omitempty"`
+	Severities      []string              `json:"severities,omitempty"`
+	Updated         DateTimeRange         `json:"updated,omitempty"`
+	SecurityFilters VulnerabilitiesFilter `json:"security_filters,omitempty"`
+	LicenseFilters  LicensesFilter        `json:"license_filters,omitempty"`
 }
 
 type CvssScore struct {
-	MinScore float32 `json:"min_score,omitempty"`
-	MaxScore float32 `json:"max_score,omitempty"`
+	MinScore float64 `json:"min_score,omitempty"`
+	MaxScore float64 `json:"max_score,omitempty"`
```

Comment:
@ns-dsinghal could you check if this change is really required? Lets avoid breaking back compat.

Reply:
yes it is required. perviously it have only few parameter all the parameter can be empty. if this is not send it end up a bad query filter.

Reply:
@ns-dsinghal whatever you said, does that apply to the change in bits for `float` here (32->64)?

Reply:
I have change to to float64 because other structure are also using float64 to hold CVE score

Reply:
restored to type float32